docs/rest_api.rst

```diff
@@ -0,0 +1,336 @@
+	      "wal_log_hints": "on",
+	      "max_wal_senders": 5
+	    },
+	    "use_pg_rewind": true
+	  },
+	  "loop_wait": 3
+	}
+
+
+Switchover and failover endpoints
+---------------------------------
+
+``POST /switchover`` or ``POST /failover``. These endpoints are very similar to each other. There are a couple of minor differences though:
+
+1. The failover endpoint allows to perform a manual failover when there are no healthy nodes, but at the same time it will not allow you to schedule a switchover.
+
+2. The switchover endpoint is the opposite. It works only when the cluster is healthy (there is a leader) and allows to schedule a switchover at a given time.
+
+
+In the JSON body of the ``POST`` request you must specify at least the ``leader`` or ``candidate`` fields and optionally the ``scheduled_at`` field if you want to schedule a switchover at a specific time.
+
+
+Example: perform a failover to the specific node:
+
+.. code-block:: bash
+
+    $ curl -s http://localhost:8009/failover -XPOST -d '{"candidate":"postgresql1"}'
+    Successfully failed over to "postgresql1"
+
+
+Example: schedule a switchover from the leader to any other healthy replica in the cluster at a specific time:
+
+.. code-block:: bash
+
+    $ curl -s http://localhost:8008/switchover -XPOST -d \
+	    '{"leader":"postgresql0","scheduled_at":"2019-09-24T12:00+00"}'
+    Switchover scheduled
+
+
+Depending on the situation the request might finish with a different HTTP status code and body. The status code **200** is returned when the switchover or failover successfully completed. If the switchover was successfully scheduled, Patroni will return HTTP status code **202**. In case something went wrong, the error status code (one of **400**, **412** or **503**) will be returned with some details in the response body. For more information please check the source code of ``patroni/api.py:do_POST_failover()`` method.
+
+The switchover and failover endpoints are used by ``patronictl switchover`` and ``patronictl failover``, respectively.
+
+
+Restart endpoint
+----------------
+
+- ``POST /restart``: You can restart Postgres on the specific node by performing the ``POST /restart`` call. In the JSON body of ``POST`` request it is possible to optionally specify some restart conditions:
+
+  - **restart_pending**: boolean, if set to ``true`` Patroni will restart PostgreSQL only when restart is pending in order to apply some changes in the PostgreSQL config.
+  - **role**: perform restart only if the current role of the node matches with the role from the POST request.
+  - **postgres_version**: perform restart only if the current version of postgres is smaller than specified in the POST request.
+  - **timeout**: how long we should wait before PostgreSQL starts accepting connections. Overrides ``master_start_timeout``.
+  - **schedule**: timestamp with time zone, schedule the restart somewhere in the future.
+
+- ``DELETE /restart``: delete the scheduled restart
```

Comment:
this is used with ``patronictl flush``, right?

Reply:
Yes, updated and pushed.